### PR TITLE
chore(identity): add resource provider check

### PR DESCRIPTION
**Pull request by xzf0587**
add resource provider check for identity

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/identity/index.ts [hunk 1]
```diff
@@ -65,6 +67,17 @@ export class IdentityPlugin implements Plugin {
     );
     this.config.location = ContextUtils.getConfigString(Constants.solution, Constants.location);
 
+    try {
+      ctx.logProvider?.info(Message.checkProvider);
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/identity/index.ts:71]: Add progress bar?

**Review by KennethBWSong — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/identity/utils/messages.ts [hunk 2]
```diff
@@ -5,7 +5,11 @@ export class Message {
   public static readonly endProvision = `[${Constants.pluginName}] end provision`;
   public static readonly provisionIdentity = `[${Constants.pluginName}] provision identity`;
   public static readonly getIdentityId = `[${Constants.pluginName}] get identity id`;
+  public static readonly checkProvider = `[${Constants.pluginName}] check SQL resource provider`;
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/identity/utils/messages.ts:8]: check identity resource provide?

File: packages/fx-core/src/plugins/resource/identity/index.ts [hunk 3]
```diff
@@ -65,6 +67,17 @@ export class IdentityPlugin implements Plugin {
...
+      const credentials = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+      const resourceManagementClient = new Providers(
+        new ResourceManagementClientContext(credentials!, this.config.azureSubscriptionId)
+      );
+      await resourceManagementClient.register(Constants.resourceProvider);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/identity/index.ts:76]: Just realized we will try to register resource provider every time? If user does not have permission to register resource provider, but the resource provider is already registered, will this operation fail?

File: packages/fx-core/src/plugins/resource/identity/utils/messages.ts (hunk 2 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/identity/utils/messages.ts]: it is fixed. Thanks.

File: packages/fx-core/src/plugins/resource/identity/index.ts (hunk 3 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/identity/index.ts:76]: We try to register the resource every time. If the register failed, we continue to provide the resource. So the permission to register resource will not interrupt the provision.

**Pull request merged**